### ISSUE TITLE
Bug: Unexpected error in conditional type

### ISSUE THREAD

**Issue by ZBAGI** [palantir/tslint#4168]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.8.4
- __Running TSLint via__: Visual Studio

#### TypeScript code being linted

```ts
import { INumberValueFilter } from "./numberValueFilter.interface";
import { IStringValueFilter } from "./stringValueFilter.interface";

export type IValueFilter<TValue> =
    TValue extends string ? IStringValueFilter :
    TValue extends number ? INumberValueFilter :
    never;
```

with `tslint.json` configuration:

```json
{
  "rulesDirectory": [
    "node_modules/codelyzer"
  ],
  "rules": {
    "arrow-return-shorthand": true,
    "callable-types": true,
    "class-name": true,
    "comment-format": [
      true,
      "check-space"
    ],
    "curly": true,
    "deprecation": {
      "severity": "warn"
    },
    "eofline": true,
    "forin": true,
    "import-blacklist": [
      true,
      "rxjs/Rx"
    ],
    "import-spacing": true,
    "indent": [
      true,
      "spaces"
    ],
    "interface-over-type-literal": true,
    "label-position": true,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "member-ordering": [
      true,
      {
        "order": [
          "static-field",
          "instance-field",
          "static-method",
          "instance-method"
        ]
      }
    ],
    "no-arg": true,
    "no-bitwise": true,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-construct": true,
    "no-debugger": true,
    "no-duplicate-super": true,
    "no-empty": false,
    "no-empty-interface": false,
    "no-eval": true,
    "no-inferrable-types": [
      true,
      "ignore-params"
    ],
    "no-misused-new": true,
    "no-non-null-assertion": true,
    "no-shadowed-variable": true,
    "no-string-literal": false,
    "no-string-throw": true,
    "no-switch-case-fall-through": true,
    "no-trailing-whitespace": true,
    "no-unnecessary-initializer": true,
    "no-unused-expression": true,
    "no-use-before-declare": true,
    "no-var-keyword": true,
    "object-literal-sort-keys": false,
    "one-line": [
      true,
      "check-open-brace",
      "check-catch",
      "check-else",
      "check-whitespace"
    ],
    "prefer-const": true,
    "quotemark": [
      true,
      "double"
    ],
    "radix": true,
    "semicolon": [
      true,
      "always"
    ],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      true,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "unified-signatures": true,
    "variable-name": false,
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-separator",
      "check-type"
    ],
    "no-output-on-prefix": true,
    "use-input-property-decorator": true,
    "use-output-property-decorator": true,
    "use-host-property-decorator": true,
    "no-input-rename": true,
    "no-output-rename": true,
    "use-life-cycle-interface": true,
    "use-pipe-transform-interface": true,
    "component-class-suffix": true,
    "directive-class-suffix": true
  }
}
```

#### Actual behavior
TSLint gives following errors:
 - `[tslint] Missing semicolon (semicolon)` for `TValue` (line 5 and 6)
 - `[tslint] unused expression, expected an assignment or function call (no-unused-expression)` for second part of `extends` (line 5, 6 and 7)

#### Expected behavior
No errors.

**Comment by Fabianopb** [comment]
Getting same errors here, but running TSLint via Atom...

**Comment by JoshuaKGoldberg** [comment]
> upgraded TS to 3.1.6

Yup, that should do it. This bug is likely caused by an older version of TypeScript not understanding how to parse newer syntax.

Related: #3808 tracks informing the user why their errors are funky.
